docs/docs/unreachable.md

```diff
@@ -0,0 +1,143 @@
+
+## DC/OS Unreachable Task Scenarios
+
+The follow scenarios a given for a task going unreachable given a default configuration of a DC/OS cluster. This also assumes that there are resources with required constraints available in the cluster for the replacement task. All marked time references are from the actual loss of an agent from the cluster.
+
+### Scenario 1: UnreachableStrategy `{0, 0}`, 1 agent becomes unresponsive for 4 minutes
+
+- 12:00:00 - Agent 1 stops responding due to a network partition (and is running one task).
+- 12:04:00 - The network partition is resolved, and agent 1 starts responding again.
+
+At a task level, Marathon is not made aware that the agent was momentarily not responding; Mesos does not report the  task as unreachable.
+
+### Scenario 2: UnreachableStrategy `{0, 0}`, 1 agent becomes unresponsive for 6 minutes
+
+- 12:00:00 - Agent 1 stops responding due to a network partition (and is running one task).
+- 12:05:00 - The Mesos master marks Agent 1 as inactive, and publishes a `TASK_UNREACHABLE` status update.
+- 12:05:03 - Shortly after, Marathon replaces 1 tasks for with the `TASK_UNREACHABLE` status was received. Record of the unreachable task is quickly expunged.
+- 12:06:00 - Agent 1 starts responding again. The Mesos master marks the agent as reachable, and sends a `TASK_RUNNING`status update.
+- 12:06:03 - Marathon kills the previously unreachable task kill.
+
+### Scenario 3: UnreachableStrategy `{0, 300}`, lose 1 node for 5 minutes; node becomes reachable after 6 minutes of original event.
+
+- 12:00:00 - Agent 1 becomes unreachable due to a network partition (and is running 1 task).
+- 12:05:00 - The Mesos master marks Agent 1 as inactive, and publishes a `TASK_UNREACHABLE` update for the task associated with it.
+- 12:05:00 - Marathon launches a replacement task for which the `TASK_UNREACHABLE` status was received. However, the currently unreachable task is not expunged, and Marathon shows 2 tasks for an application with a target instance count of 1.
+- 12:06:00 - Agent 1 becomes starts responding again, and the Mesos master marks it as active. The task associated with agent 1 is still running, so a `TASK_RUNNING` status update is published for it.
+- Between 12:10:00 to 12:20:00 - the reachable task is killed, and Marathon reports 1 of 1 for application.
```

Comment:
`the reachable task is killed`: I believe this is false (please correct me if I'm wrong): according to the default  `"killSelection": "YoungestFirst"` the replacement task started at 12:05:00 should be killed here.

Reply:
the reachable task. Which do you mean? The original task or the replacement task?